### PR TITLE
disable non-editable fields when editing a lock

### PR DESCRIPTION
**Pull request by cellog**
# Description

When editing a lock, we only want key price to be editable (currently). This PR disables all other fields and dims the text as a visual indicator.

screen shot 2019-01-16 at 6 59 45 pm
screen shot 2019-01-16 at 6 59 39 pm

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [X] If my code involves visual changes, I am adding applicable screenshots to this thread

PS: [Read how to write the perfect pull request](https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/)

### CODE REVIEW

File: unlock-app/src/components/creator/CreatorLockForm.js [hunk 1]
```diff
@@ -135,6 +135,8 @@ export class CreatorLockForm extends React.Component {
   }
 
   render() {
+    const { address } = this.props
+    const disableEdit = address.length === 42
```
Comment (julien51) [inline unlock-app/src/components/creator/CreatorLockForm.js]: Do you mind clarify what is 42 here and why this matters?

File: unlock-app/src/stories/creator/CreatorLockForm.stories.js [hunk 2]
```diff
@@ -22,17 +22,13 @@ storiesOf('CreatorLockForm', module)
     // TODO: implement this
     const lock = {
       keyPrice: '10000000000000000000',
-      expirationDuration: '172800',
+      expirationDuration: 172800,
```
Comment (julien51) [inline unlock-app/src/stories/creator/CreatorLockForm.stories.js]: Not sure why this change is part of the PR? Could you please clarify?

File: unlock-app/src/stories/creator/CreatorLockForm.stories.js [hunk 3]
```diff
@@ -22,17 +22,13 @@ storiesOf('CreatorLockForm', module)
...
       address: '0xab7c74abc0c4d48d1bdad5dcb26153fc8780f83e',
       name: 'Existing Lock',
     }
     return (
-      <CreatorLockForm
-        lock={lock}
-        hideAction={() => {}}
-        createLock={() => {}}
-      />
+      <CreatorLockForm hideAction={() => {}} createLock={() => {}} {...lock} />
```
Comment (julien51) [inline unlock-app/src/stories/creator/CreatorLockForm.stories.js]: I am not sure why you change the `lock={lock}` into `{...lock}`? I think it is probably cleaner to keep the object "as-is" rather than pass its value in order to differentiate between the actual values and the other props which are not lock properties, such as `hideAction` or `createLock`.

File: unlock-app/src/components/creator/CreatorLockForm.js (hunk 1 above)
Comment (cellog) [reply unlock-app/src/components/creator/CreatorLockForm.js]: the difference between a new lock and a full lock is the new lock uses a `uniqid()` as its address, which is length 18. A standard lock address is 42. It may be better to explicitly pass in a "editing" prop for this.

File: unlock-app/src/stories/creator/CreatorLockForm.stories.js (hunk 2 above)
Comment (cellog) [reply unlock-app/src/stories/creator/CreatorLockForm.stories.js]: it generated a console warning, which I assumed was due to code changes in this PR since I hadn't seen it before

File: unlock-app/src/stories/creator/CreatorLockForm.stories.js (hunk 3 above)
Comment (cellog) [reply unlock-app/src/stories/creator/CreatorLockForm.stories.js]: this has been part of the API for CreatorLockForm since #1029 

If you'd like, it would be relatively trivial to change this post-this merge in another PR, I'm happy to open an issue and self-assign.

File: unlock-app/src/components/creator/CreatorLockForm.js (hunk 1 above)
Comment (julien51) [reply unlock-app/src/components/creator/CreatorLockForm.js]: I think this is too brittle and we should not use the address to determine that a lock is `pending` because we recently discovered that we could get the address of the lock even though the transaction was not mined previously.

There is actually a `pending` value on the lock object which you can re-use here?

Comment (cellog) [reply unlock-app/src/components/creator/CreatorLockForm.js]: excellent! That is what I was looking for, thanks

File: unlock-app/src/stories/creator/CreatorLockForm.stories.js (hunk 2 above)
Comment (julien51) [reply unlock-app/src/stories/creator/CreatorLockForm.stories.js]: Let's remove this from this PR then.

File: unlock-app/src/stories/creator/CreatorLockForm.stories.js (hunk 3 above)
Comment (julien51) [reply unlock-app/src/stories/creator/CreatorLockForm.stories.js]: Ha indeed, it would be much cleaned to use the `lock` object. Please open an issue and asisgn it to yourself and work on that after all is done on this front.

Comment (cellog) [reply unlock-app/src/stories/creator/CreatorLockForm.stories.js]: #1075

File: unlock-app/src/stories/creator/CreatorLockForm.stories.js (hunk 2 above)
Comment (cellog) [reply unlock-app/src/stories/creator/CreatorLockForm.stories.js]: ok

**Comment by cellog** [comment]
@julien51 ready for final (hopefully) review :)

**Comment by cellog** [comment]
hang on - the console error I referred to is introduced by this PR, so I need to add that back in

File: unlock-app/src/stories/creator/CreatorLockForm.stories.js (hunk 2 above)
Comment (cellog) [reply unlock-app/src/stories/creator/CreatorLockForm.stories.js]: this is needed because we are using `{...lock}` whereas prior, the lock was not being passed in at all, and props require a number. It was an error that was present but never triggered, so we do need it after all

Comment (julien51) [reply unlock-app/src/stories/creator/CreatorLockForm.stories.js]: ok that works.
